decoder_test.go

```diff
@@ -221,20 +212,14 @@ func TestReadUint64(t *testing.T) {
 		}
 
 		number, err := d.readUint64()
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
```

Comment:
Please consider using assert.Equal and put expected and actual arguments in a correct order.